src/containers/MediaElementContainer.js

```diff
@@ -2,81 +2,81 @@ import React, { Component } from 'react';
 import MediaElement from '../components/MediaElement';
 import PropTypes from 'prop-types';
 import ErrorMessage from '../components/ErrorMessage';
-import manifesto from 'manifesto.js';
+import {
+  getChoiceItems,
+  getSources,
+  getMediaType
+} from '../services/iiif-parser';
+import { connect } from 'react-redux';
 
 class MediaElementContainer extends Component {
   state = {
     manifest: this.props.manifest,
     ready: false,
     sources: [],
     mediaType: null,
+    canvasIndex: 0,
     error: null
   };
 
   componentDidMount() {
     const { manifest } = this.state;
-    let choiceItems = [];
-
-    try {
-      choiceItems = manifesto
-        .create(manifest)
-        .getSequences()[0]
-        .getCanvases()[0]
-        .getContent()[0]
-        .getBody();
-    } catch (e) {}
-
-    this.prepSources(choiceItems);
-  }
-
-  getSources(choiceItems) {
-    const sources = choiceItems.map(item => {
-      return {
-        src: item.id,
-        // TODO: make type more generic, possibly use mime-db
-        format: item.getFormat().value
-      };
-    });
-    return sources;
-  }
-
-  getMediaType(choiceItems) {
-    let allTypes = choiceItems.map(item => item.getType().value);
-    let uniqueTypes = allTypes.filter((t, index) => {
-      return allTypes.indexOf(t) === index;
-    });
-    if (uniqueTypes.length === 1) {
-      return uniqueTypes[0];
-    }
-    // Default type if there are different types
-    return 'audio';
-  }
-
-  prepSources(choiceItems) {
+    // Get the first canvas in manifest
+    let choiceItems = getChoiceItems(manifest, 0);
     if (choiceItems.length === 0) {
       this.setState({
         error: 'No media choice items found in manifest'
```

Comment:
If this error message is user facing, would it make sense to change this error message and the one on line 49 to be `No media sources found`?  It is less specific but avoids jargon.  If it isn't user facing, then maybe `No media choice items found for canvas` would be more accurate.